src/stream/src/executor/mview/materialize.rs

```diff
@@ -406,20 +416,24 @@ impl<S: StateStore> std::fmt::Debug for MaterializeExecutor<S> {
 }
 
 /// A cache for materialize executors.
-pub struct MaterializeCache {
+pub struct MaterializeCache<SD> {
     data: ExecutorCache<Vec<u8>, Option<CompactedRow>>,
+    _serde: PhantomData<SD>,
 }
 
-impl MaterializeCache {
+impl<SD: ValueRowSerde> MaterializeCache<SD> {
     pub fn new(watermark_epoch: AtomicU64Ref) -> Self {
         let cache = ExecutorCache::new(new_unbounded(watermark_epoch));
-        Self { data: cache }
+        Self {
+            data: cache,
+            _serde: PhantomData,
+        }
     }
 
     pub async fn handlle_conflict<'a, S: StateStore>(
         &mut self,
         buffer: MaterializeBuffer,
-        table: &StateTable<S>,
+        table: &StateTableInner<S, SD>,
```

Comment:
can we extract a trait for the state table such as `StateTableRead` and change the type to `impl StateTableRead` to  erase the `S` and `SD` here :thinking:

Reply:
Might be a good idea. We can do this refactor in the future.